### PR TITLE
[Architecture] Kotlin VanillaPacksTab

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

Completely rewrite `VanillaPacksTab` with support for
- Relocalization
- Kotlin
- ViewModel
- Async computation of various UI elements

Now that the main rewrite has been complete, I need to first finish polishing up issues with the async to what the previous PR had (it working).

### Testing

- [x] Create Server
  - [x] With mod loader
- [x] Create Instance
  - [x] With mod loader

### Related Issues

#606 
#695 
Old #659

### CODE REVIEW

File: src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+                MinecraftManager.getMinecraftVersion(selectedMinecraftVersion),
+                selectedLoaderVersion,
+                description
+            )
+            installable.instanceName = name
+            installable.isReinstall = false
+            installable.isServer = isServer
+            val success = installable.startInstall()
+            if (success) {
+                // TODO Reset server
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt:215]: Is this still a TODO?

File: src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+    override fun setSelectedMinecraftVersion(newVersion: String?) {
+        selectedMinecraftVersionFlow.value = newVersion
+    }
+
+    override val showFabricOption: Boolean by lazy { ConfigManager.getConfigItem("loaders.fabric.enabled", true) }
+
+    override val showForgeOption: Boolean by lazy { ConfigManager.getConfigItem("loaders.forge.enabled", true) }
+    override val showLegacyFabricOption: Boolean by lazy {
+        ConfigManager.getConfigItem(
+            "loader.legacyfabric.enabled",
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt:350]: `loaders.legacyfabric.enabled`

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]

File: src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt (hunk 1 above)
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/viewmodel/impl/VanillaPacksViewModel.kt:215]: I believe this was what we discussed on discord about resetting the UI after an install is done. So no, I will remove the TODO.

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty A small ping, just in case you did not get a re-review email.

**Review by RyanTheAllmighty — APPROVED** [review]

**Pull request merged**